### PR TITLE
initial commit on static file feature

### PR DESCRIPTION
**Pull request by shawn-mcginty**
this hasn't been fully tested yet, but this is my general idea.  Using `Lwt_stream` to handle static files (or really any large amount of data).  And helpers for static file and mime types.

### CODE REVIEW

File: morph_core/src/MimeTypes.re [hunk 1]
```diff
@@ -0,0 +1,1003 @@
+let getExtension = filename => {
```
Comment (anuragsoni) [inline morph_core/src/MimeTypes.re]: there is an opam package that deals with mime types https://github.com/mirage/ocaml-magic-mime

Comment (shawn-mcginty) [reply morph_core/src/MimeTypes.re]: good catch! switched to that guy.

**Comment by shawn-mcginty** [comment]
This is now ready to test, it's working for me.

**Comment by ulrikstrid** [comment]
I looked a bit at how Cohttp does it and they just handle a `Body.t` as a binary thing so it can basically be anything and the user has to do the conversion. I think we should do the same but I think this is good to go for now.

Thank you for this!

**Pull request merged**